src/core/ComposableElement.php

```diff
@@ -357,7 +357,7 @@ private static function isSpreadKey(string $key): bool {
         $value = $this->validateAttributeValue($attr, $value);
       }
     } else {
-      $value = (string)$value;
+      $value = $value;
```

Comment:
Both before and after my changes `data-foo={123}` becomes `data-foo="123"` because of the implicit string cast due to calling htmlspecialchars()

```
hphpd> var_dump(htmlspecialchars(123));
string(3) "123"
```

Added another test case demonstrating this